File: aulas/python-básico/ack.py
# https://www.youtube.com/watch?v=i7sm9dzFtEI
def ack(m, n):
    if m == 0:
        return n + 1
    elif n == 0:
        return ack(m - 1, n + 1)
    else:
        return ack(m - 1, ack(m, n - 1))


for m in range(6):
    for n in range(6):
        print(f"{m=}, {n=}")
        print(f"   ack={ack(m, n)}")

File: aulas/python-básico/ack.c
#include <stdio.h>

int ack(m, n)
int m, n;
{
    int ans;
    if (m == 0)
        ans = n + 1;
    else if (n == 0)
        ans = ack(m - 1, 1);
    else
        ans = ack(m - 1, ack(m, n - 1));
    return (ans);
}
